To: vim_dev@googlegroups.com
Subject: Patch 7.3.378
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.378
Problem:    When cross-compiling the check for uint32_t fails.
Solution:   Only give a warning message. (Maksim Melnikau)
Files:      src/configure.in, src/auto/configure


*** ../vim-7.3.377/src/configure.in	2011-12-08 15:17:28.000000000 +0100
--- src/configure.in	2011-12-14 19:18:36.000000000 +0100
***************
*** 3283,3289 ****
  }],
  AC_MSG_RESULT(ok),
  AC_MSG_ERROR([WRONG!  uint32_t not defined correctly.]),
! AC_MSG_ERROR([could not compile program using uint32_t.]))
  
  dnl Check for memmove() before bcopy(), makes memmove() be used when both are
  dnl present, fixes problem with incompatibility between Solaris 2.4 and 2.5.
--- 3283,3289 ----
  }],
  AC_MSG_RESULT(ok),
  AC_MSG_ERROR([WRONG!  uint32_t not defined correctly.]),
! AC_MSG_WARN([cannot check uint32_t when cross-compiling.]))
  
  dnl Check for memmove() before bcopy(), makes memmove() be used when both are
  dnl present, fixes problem with incompatibility between Solaris 2.4 and 2.5.
*** ../vim-7.3.377/src/auto/configure	2011-12-08 15:17:28.000000000 +0100
--- src/auto/configure	2011-12-14 19:20:55.000000000 +0100
***************
*** 11777,11783 ****
  { $as_echo "$as_me:${as_lineno-$LINENO}: checking uint32_t is 32 bits" >&5
  $as_echo_n "checking uint32_t is 32 bits... " >&6; }
  if test "$cross_compiling" = yes; then :
!   as_fn_error "could not compile program using uint32_t." "$LINENO" 5
  else
    cat confdefs.h - <<_ACEOF >conftest.$ac_ext
  /* end confdefs.h.  */
--- 11777,11784 ----
  { $as_echo "$as_me:${as_lineno-$LINENO}: checking uint32_t is 32 bits" >&5
  $as_echo_n "checking uint32_t is 32 bits... " >&6; }
  if test "$cross_compiling" = yes; then :
!   { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: cannot check uint32_t when cross-compiling." >&5
! $as_echo "$as_me: WARNING: cannot check uint32_t when cross-compiling." >&2;}
  else
    cat confdefs.h - <<_ACEOF >conftest.$ac_ext
  /* end confdefs.h.  */
*** ../vim-7.3.377/src/version.c	2011-12-14 15:32:44.000000000 +0100
--- src/version.c	2011-12-14 19:21:08.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     378,
  /**/

-- 
            ### Hiroshima 45, Chernobyl 86, Windows 95 ###

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
